exonum/src/runtime/dispatcher/schema.rs

```diff
@@ -149,20 +149,42 @@ impl Schema<&Fork> {
             return Err(Error::ServiceIdExists);
         }
 
-        let id = spec.id;
-        let name = spec.name.clone();
+        let instance_id = spec.id;
+        let instance_name = spec.name.clone();
+        let pending_status = InstanceStatus::Active;
+
         instances.put(
-            &name,
+            &instance_name,
             InstanceState {
-                spec: spec.clone(),
-                status: InstanceStatus::Pending,
+                spec,
+                status: None,
+                pending_status: Some(pending_status),
             },
         );
-        instance_ids.put(&id, name);
-        self.pending_instances().push(spec);
+        self.pending_instances().put(&instance_name, pending_status);
+        instance_ids.put(&instance_id, instance_name);
         Ok(())
     }
 
+    /// Adds information about stopping service instance to the schema.
+    pub(crate) fn initiate_stopping_service(&mut self, instance_id: InstanceId) {
+        let mut instances = self.instances();
+
+        let instance_name = self
+            .instance_ids()
+            .get(&instance_id)
+            .expect("BUG: attempt to stop a nonexistent service instance.");
```

Comment:
I'd mention in the description that this method assumes the input data to be correct.

Reply:
Do you think that `BUG` prefix is not enough?

Reply:
I think it's better to know that function panics before it actually panics :)

Reply:
For example, such a note will be useful for implementers of custom `Supervisor` service.

Reply:
I'll write "panic" clause in the method documentation.